wolfcrypt/src/aes.c

```diff
@@ -3085,8 +3085,17 @@ int wc_AesSetIV(Aes* aes, const byte* iv)
     int wc_AesCbcEncrypt(Aes* aes, byte* out, const byte* in, word32 sz)
     {
         int ret = 0;
-        word32 blocks = (sz / AES_BLOCK_SIZE);
         CRYP_HandleTypeDef hcryp;
+        word32 blocks = (sz / AES_BLOCK_SIZE);
+
+#ifdef WOLFSSL_AES_CBC_ALIGNMENT_CHECKS
+        if (sz % AES_BLOCK_SIZE) {
+            return BAD_ALIGN_E;
+        }
+#else
+        if (blocks == 0)
```

Comment:
I like the addition of early return "success" with zero length. I see no issue with this. Any reason not to keep this logic for the non `WOLFSSL_AES_CBC_ALIGNMENT_CHECKS` case?

Reply:
Done.  I was saving a cycle in the `WOLFSSL_AES_CBC_ALIGNMENT_CHECKS` path, but that was probably penny wise and pound foolish :-)